### ISSUE TITLE
haddock-api keeps rebuilding

### ISSUE THREAD

**Issue by AlexeyRaga** [haskell/haskell-ide-engine#540]
Building HIE always re-builds `haddock-api`, even if nothing has changed (subsequent `stack build` commands).

For some reason, it is **VERY** slow. Does anyone know why it happens and what can be done about it?

```
haddock-library-1.4.4: unregistering
haddock-library-1.4.4: configure
haddock-library-1.4.4: build
haddock-library-1.4.4: copy/register
haddock-api-2.18.1: configure
haddock-api-2.18.1: build
haddock-api-2.18.1: copy/register
```

**Comment by alanz** [comment]
It's a stack thing.

I develop with `cabal new-build` (using cabal 2.2), and this problem does not occur.

Note, I have not checked if it still happens with stack 1.7.1

**Comment by mgsloan** [comment]
This should be resolved in stack 1.7.1 https://github.com/commercialhaskell/stack/pull/3955

The issue was that haddock-api added a use of a sublibrary and stack's rebuild logic hadn't yet been updated to handle that case properly.
